### PR TITLE
chore: Update Windows VHD image to 5B with some fixes

### PR DESCRIPTION
**Pull request by AbelHu**
**Reason for Change**:

Update Windows VHD image to 5B with some fixes

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:
Cherry pick below commits from master branch:
&nbsp;&nbsp;fix: Expose error details in Windows CSE (#3159)
&nbsp;&nbsp;feat: Updating Windows VHDs to include May patches (#3263)
&nbsp;&nbsp;fix: the os disk size restriction is changed in azure. (#3043)
&nbsp;&nbsp;refactor: Create signable PS scripts (#3015)
&nbsp;&nbsp;fix: don't hardcode pause image for containerD on Windows (#3158)
&nbsp;&nbsp;chore: update Windows pause image (#3210)
&nbsp;&nbsp;fix: USER_ASSIGNED_IDENTITY_ID is empty in azure.json (#3254)
&nbsp;&nbsp;fix: use full userAssignedID reference in ARM template

### CODE REVIEW

File: pkg/api/azenvtypes.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ var (
 		AzureCNIImageBase:                "mcr.microsoft.com/containernetworking/",
 		MCRKubernetesImageBase:           "mcr.microsoft.com/",
 		EtcdDownloadURLBase:              "mcr.microsoft.com/oss/etcd-io/",
-		KubeBinariesSASURLBase:           "https://acs-mirror.azureedge.net/kubernetes/",
+		KubeBinariesSASURLBase:           "https://kubernetesartifacts.azureedge.net/kubernetes/",
```
Comment (xuto2) [inline pkg/api/azenvtypes.go:94]: no, we need to stick toacs-mirror

File: pkg/api/azenvtypes.go [hunk 2]
```diff
@@ -288,9 +288,9 @@ var (
 			EtcdDownloadURLBase:              "mcr.microsoft.com/oss/etcd-io/",
 			KubeBinariesSASURLBase:           DefaultKubernetesSpecConfig.KubeBinariesSASURLBase,
 			WindowsTelemetryGUID:             DefaultKubernetesSpecConfig.WindowsTelemetryGUID,
-			CNIPluginsDownloadURL:            "https://mirror.azk8s.cn/kubernetes/containernetworking-plugins/cni-plugins-amd64-" + CNIPluginVer + ".tgz",
-			VnetCNILinuxPluginsDownloadURL:   "https://mirror.azk8s.cn/kubernetes/azure-container-networking/azure-vnet-cni-linux-amd64-" + AzureCniPluginVerLinux + ".tgz",
-			VnetCNIWindowsPluginsDownloadURL: "https://mirror.azk8s.cn/kubernetes/azure-container-networking/azure-vnet-cni-windows-amd64-" + AzureCniPluginVerWindows + ".zip",
+			CNIPluginsDownloadURL:            "https://mirror.azk8s.cn/kubernetes/containernetworking-plugins/cni-plugins-linux-amd64-" + CNIPluginVer + ".tgz",
+			VnetCNILinuxPluginsDownloadURL:   "https://kubernetesartifacts.blob.core.chinacloudapi.cn/azure-cni/" + AzureCniPluginVerLinux + "/binaries/azure-vnet-cni-linux-amd64-" + AzureCniPluginVerLinux + ".tgz",
```
Comment (xuto2) [inline pkg/api/azenvtypes.go:292]: I believe *.azk8s.cn is the approved egress for china.

https://docs.microsoft.com/en-us/azure/aks/limit-egress-traffic#egress-traffic-overview

Let's revert this change

File: pkg/api/azenvtypes.go (hunk 1 above)
Comment (AbelHu) [reply pkg/api/azenvtypes.go:94]: The PR "build: fetch k8s Windows .zip from kubernetesartifacts storage #2655" introduced this change. This is only for Windows to download init zip package for Windows.

Comment (AbelHu) [reply pkg/api/azenvtypes.go:94]: https://github.com/tamilmani1989/aks-engine/blob/30ecbad52f8802f08a2115cfd1950a5e89a1b93b/pkg/engine/params_k8s.go#L126-L133
```
			if properties.HasWindows() {
				// Kubernetes packages as zip file as created by scripts/build-windows-k8s.sh
				// will be removed in future release as if gets phased out (https://github.com/Azure/aks-engine/issues/3851)
				kubeBinariesSASURL := kubernetesConfig.CustomWindowsPackageURL
				if kubeBinariesSASURL == "" {
					kubeBinariesSASURL = cloudSpecConfig.KubernetesSpecConfig.KubeBinariesSASURLBase + k8sComponents["windowszip"]
				}
				addValue(parametersMap, "kubeBinariesSASURL", kubeBinariesSASURL)
```

File: pkg/api/azenvtypes.go (hunk 2 above)
Comment (AbelHu) [reply pkg/api/azenvtypes.go:292]: Good catch. Will remove the commit "chore: update zip name for mooncake mirror (#3237)"

Comment (AbelHu) [reply pkg/api/azenvtypes.go:292]: File an issue #3279

File: pkg/api/azenvtypes.go (hunk 1 above)
Comment (xuto2) [reply pkg/api/azenvtypes.go:94]: https://docs.microsoft.com/en-us/azure/aks/limit-egress-traffic

We deliberately keep using acs-mirror to not break our egress promise.

Comment (AbelHu) [reply pkg/api/azenvtypes.go:94]: Will update it

File: pkg/api/defaults.go [hunk 3]
```diff
@@ -496,6 +496,14 @@ func (cs *ContainerService) setOrchestratorDefaults(isUpgrade, isScale bool) {
...
 				}
 			}
+			// Ensure that all VMSS pools have SinglePlacementGroup set to false in Standard LB cluster scenarios
+			if profile.AvailabilityProfile == VirtualMachineScaleSets && profile.SinglePlacementGroup == nil {
+				if cs.Properties.OrchestratorProfile.KubernetesConfig.LoadBalancerSku == StandardLoadBalancerSku {
+					profile.SinglePlacementGroup = to.BoolPtr(false)
+				} else {
+					profile.SinglePlacementGroup = to.BoolPtr(DefaultSinglePlacementGroup)
+				}
+			}
```
Comment (xuto2) [inline pkg/api/defaults.go:506]: This is a big behavior change - can we remove this for safety?

Comment (xuto2) [reply pkg/api/defaults.go:506]: I can send email to discuss the expected behavior for aks

Comment (AbelHu) [reply pkg/api/defaults.go:506]: Agree with that this can be removed from this PR.
